pkg/asset/installconfig/aws/regions.go

```diff
@@ -37,3 +41,20 @@ func IsKnownRegion(region string, architecture types.Architecture) bool {
 	}
 	return false
 }
+
+// IsKnownAWSRegion gets all the supported AWS regions currently using the SDK session call and
+// checks if the given region is present in the list of regions returned.
+func IsKnownAWSRegion(region string) (bool, error) {
+	session := ec2.New(session.New(), &aws.Config{Region: aws.String("us-east-1")})
```

Comment:
Hmm. I didn't consider that we need to hit a known region in order to list the available regions. We cannot use "us-east-1" when the partition requested is not the standard partition. We would need to find the partition for the selected region first, and then use a region in that partition to validate that the region exists. There would also need to be consideration for whether the user provided an override for the ec2 service endpoint.

I am coming back around to the idea of running a test query against the selected region with a short timeout rather than querying for whether the region exists directly.

Reply:
Yeah that works locally but the tests are failing with almost all regions. Not really sure but is it an awscred file error?